File: app/src/main/java/todo/list/ToDoListViewModel.java
package todo.list;

import android.app.Application;
import android.arch.lifecycle.AndroidViewModel;
import android.arch.lifecycle.LiveData;

import java.util.List;

import todo.model.ToDo;
import todo.persistence.ToDoDatabase;

public class ToDoListViewModel extends AndroidViewModel {

    private final LiveData<List<ToDo>> toDoList;

    private ToDoDatabase database;

    public ToDoListViewModel(Application application) {
        super(application);

        database = ToDoDatabase.getDatabase(application);
        toDoList = database.toDoDao().getAll();
    }

    public LiveData<List<ToDo>> getToDoList() {
        return toDoList;
    }
}

File: app/src/main/kotlin/todo/list/ToDoListViewModel.kt
package todo.list

import android.app.Application
import android.arch.lifecycle.AndroidViewModel
import android.arch.lifecycle.LiveData
import todo.model.ToDo
import todo.persistence.ToDoDatabase

class ToDoListViewModel(application: Application) : AndroidViewModel(application) {

    val toDoList: LiveData<List<ToDo>>

    private val database: ToDoDatabase = ToDoDatabase.getDatabase(application)

    init {
        toDoList = database.toDoDao().all
    }
}
